src/dbnode/storage/shard.go

```diff
@@ -2538,6 +2546,70 @@ func (s *dbShard) Repair(
 	return repairer.Repair(ctx, nsCtx, nsMeta, tr, s)
 }
 
+func (s *dbShard) AggregateTiles(
+	ctx context.Context,
+	reader fs.DataFileSetReader,
+	sourceNsID ident.ID,
+	sourceShard databaseShard,
+	opts AggregateTilesOptions,
+	wOpts series.WriteOptions,
+) error {
+	latestSourceVolume, err := sourceShard.latestVolume(opts.Start)
```

Comment:
Won't the time range of the request sometimes span multiple index blocks? 

I think some metrics here on the amount of work its doing would be helpful as well.

Reply:
Already added a check to ensure that the writes are limited to a single target block. For source blocks (which will be smaller), we will support reading from multiple blocks, but that is being added in a different PR.

Reply:
Regarding the second comment, @gediminasgu I see you added some write error metrics, perhaps there should be write success metrics as well?